### PR TITLE
Example broken test

### PR DESCRIPTION
**Pull request by igorsergiichuk**
## Overview

| Q                 | A
| ----------------- | ---
| Bugfix?           | yes
| New feature?      | no
| Breaking Changes? | no
| Fixed issues      | changed lots of typos in SupplierModelTest

## The problem

There was lots of typos so that the UnitTest was broke constantly

## How I resolved it

SupplierModelTest.php, Supplier.php, SupplierCategory.php was added

## How to verify it

UnitTests

Notify the following people: @darrynten

### CODE REVIEW

File: tests/SageOne/Models/SupplierModelTest.php [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+            $this->assertEquals($model->defaultTaxTypeId, 1);
+            $this->assertInstanceOf(TaxType::class, $model->defaultTaxType);
+            $this->assertEquals($model->defaultTaxType->id, 1);
+            $this->assertEquals($model->defaultTaxType->name, 'sample string 2');
+            $this->assertEquals($model->defaultTaxType->percentage, 3.1);
+            $this->assertTrue($model->defaultTaxType->isDefault);
+            $this->assertTrue($model->defaultTaxType->hasActivity);
+            $this->assertTrue($model->defaultTaxType->isManualTax);
+            $this->assertEquals($model->defaultTaxType->created->format('Y-m-d'), '2017-07-04');
+            $this->assertEquals($model->defaultTaxType->modified->format('Y-m-d'), '2017-07-04');
```
Comment (darrynten) [inline tests/SageOne/Models/SupplierModelTest.php:361]: You're missing a lot of tests here, you must test every attribute

**Review by darrynten — CHANGES_REQUESTED** [review]

File: tests/SageOne/Models/SupplierModelTest.php (hunk 1 above)
Comment (igorsergiichuk) [reply tests/SageOne/Models/SupplierModelTest.php:361]: fixed

File: tests/SageOne/Models/SupplierModelTest.php [hunk 2]
```diff
@@ -0,0 +1,601 @@
...
+            $this->assertEquals($model2->defaultTaxType->modified->format('Y-m-d'), '2017-07-04');
+            $this->assertEquals($model2->dueDateMethodId, 1);
+            $this->assertEquals($model2->dueDateMethodValue, 1);
+            $this->assertEquals($model2->id, 33);
+
+            $this->assertCount(49, json_decode($model1->toJson(), true));
+            $this->assertCount(49, json_decode($model2->toJson(), true));
+        });
+    }
+
```
Comment (darrynten) [inline tests/SageOne/Models/SupplierModelTest.php:523]: `testSave` is missing from the tests

**Review by darrynten — CHANGES_REQUESTED** [review]

File: tests/SageOne/Models/SupplierModelTest.php [hunk 3]
```diff
@@ -0,0 +1,712 @@
...
+            $this->assertEquals($model->defaultTaxType->modified->format('Y-m-d'), '2017-07-04');
+            $this->assertEquals($model->dueDateMethodId, 1);
+            $this->assertEquals($model->dueDateMethodValue, 1);
+
+            $objArray = json_decode($model->toJson(), true);
+            $this->assertCount(49, $objArray);
+        });
+    }
+
+    public function testAttributes()
```
Comment (darrynten) [inline tests/SageOne/Models/SupplierModelTest.php]: Can you move this test above the inject test, makes more sense to see the definition first

File: src/Models/Supplier.php [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+     *
+     * It covers any call documented with these methods
+     *
+     * GET Model/Get
+     * GET Model/Get/{id}
+     * DELETE Model/Delete/{id}
+     * POST Model/Save
+     *
+     * You must enable the supported combination of the above 4 calls if they
+     * are present in the documentation
```
Comment (darrynten) [inline src/Models/Supplier.php]: This is boilerplate comment, you need to replace with a proper comment block ;)

File: src/Models/SupplierCategory.php [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+ */
+
+namespace DarrynTen\SageOne\Models;
+
+use DarrynTen\SageOne\BaseModel;
+
+/**
+ * Account Category
+ *
+ * Details on writable properties for Account:
```
Comment (darrynten) [inline src/Models/SupplierCategory.php]: Account should be SupplierCategory

File: src/Models/SupplierCategory.php [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+ * hhttps://accounting.sageone.co.za/api/1.1.2/#bookmark_Supplier
```
Comment (darrynten) [inline src/Models/SupplierCategory.php]: This should be link to SupplierCategory

File: src/Models/SupplierCategory.php [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+     * All must include a type, whether or not it's nullable, and whether or
+     * not it's readonly.
+     *
+     * NB: Naming convention for keys is to lowercase the first character of the
+     * field returned by Sage (they use PascalCase and we use camelCase)
+     *
+     * `ID` is automatically converted to `id`
+     *
+     * Details on writable properties for Account:
+     * https://accounting.sageone.co.za/api/1.1.2/#bookmark_Supplier
```
Comment (darrynten) [inline src/Models/SupplierCategory.php]: Wrong link and replace boilerplate comment

File: src/Models/SupplierCategory.php [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+     *
+     *
```
Comment (darrynten) [inline src/Models/SupplierCategory.php]: Too many blank lines, must only have one

**Review by darrynten — CHANGES_REQUESTED** [review]
Small change requests

File: src/Models/SupplierCategory.php (hunk 8 above)
Comment (igorsergiichuk) [reply src/Models/SupplierCategory.php]: exterminated !

File: src/Models/Supplier.php [hunk 9]
```diff
@@ -0,0 +1,306 @@
...
+
+/**
+ * Supplier Model
+ *
+ */
+class Supplier extends BaseModel
+{
+    /**
+     * The API Endpoint
+     * @var string $endpoint
```
Comment (darrynten) [inline src/Models/Supplier.php]: There should be an empty line between the comment and the `@var` line

**Review by darrynten — CHANGES_REQUESTED** [review]

File: src/Models/Supplier.php [hunk 10]
```diff
@@ -0,0 +1,306 @@
...
+     */
+    protected $endpoint = 'Supplier';
+
+    /**
+     * Defines all possible fields.
+     *
+     * This maps to all the public properties you added above
```
Comment (darrynten) [inline src/Models/Supplier.php]: This is also a boilerplate comment

**Review by darrynten — CHANGES_REQUESTED** [review]

File: src/Models/Supplier.php (hunk 10 above)
Comment (igorsergiichuk) [reply src/Models/Supplier.php]: done!

File: tests/SageOne/Models/SupplierModelTest.php (hunk 3 above)
Comment (igorsergiichuk) [reply tests/SageOne/Models/SupplierModelTest.php]: no problem

File: src/Models/Supplier.php [hunk 11]
```diff
@@ -0,0 +1,305 @@
...
+ */
+
+namespace DarrynTen\SageOne\Models;
+
+use DarrynTen\SageOne\BaseModel;
+use DarrynTen\SageOne\Models\ModelCollection;
+
+/**
+ * Supplier Model
+ *
```
Comment (darrynten) [inline src/Models/Supplier.php:18]: You're missing the link to the docs for supplier

You need to link to `https://accounting.sageone.co.za/api/1.1.2/Help#bookmark_Supplier`

**Review by darrynten — CHANGES_REQUESTED** [review]

File: src/Models/Supplier.php (hunk 4 above)
Comment (igorsergiichuk) [reply src/Models/Supplier.php]: deleted

File: src/Models/SupplierCategory.php (hunk 5 above)
Comment (igorsergiichuk) [reply src/Models/SupplierCategory.php]: done

File: src/Models/SupplierCategory.php (hunk 6 above)
Comment (igorsergiichuk) [reply src/Models/SupplierCategory.php]: done

File: src/Models/SupplierCategory.php (hunk 7 above)
Comment (igorsergiichuk) [reply src/Models/SupplierCategory.php]: done

**Review by darrynten — APPROVED** [review]

**Pull request merged**